tree/tree/test/TTreeTruncatedDatatypes.cxx

```diff
@@ -57,3 +59,42 @@ TEST(TTreeTruncatedDatatypes, float16double32leaves)
    f.Close();
    gSystem->Unlink(ofileName);
 }
+
+// https://its.cern.ch/jira/browse/ROOT-10149
+TEST(TTreeTruncatedDatatypes, LeafCounter)
+{
+   {
+      const auto ofileName = "leafcounter10149.root";
+      TFile f(ofileName, "RECREATE");
+      TTree *t = new TTree("t", "t");
+      int n;
+      Double32_t arr[64];
+      t.Branch("n", &n);
+      t.Branch("arr", arr, "arr[n]/d[0,0,10]");
```

Comment:
We need to also test (if not done elsewhere) the case with 'just' the `/d[0,0,0]` (i.e. no array) as well as `arr[16]/d[0,0,10]`